DatadogCrashReporting/Sources/CrashReportingFeature.swift

```diff
@@ -31,59 +28,57 @@ internal final class CrashReportingFeature: DatadogFeature {
         messageReceiver: FeatureMessageReceiver,
         telemetry: Telemetry
     ) {
-        self.queue = DispatchQueue(
-            label: "com.datadoghq.crash-reporter",
-            target: .global(qos: .utility)
-        )
         self.plugin = crashReportingPlugin
         self.sender = sender
         self.crashContextProvider = crashContextProvider
         self.messageReceiver = messageReceiver
         self.telemetry = telemetry
 
         // Inject current `CrashContext`
-        if let context = crashContextProvider.currentCrashContext {
-            inject(currentCrashContext: context)
+        crashContextProvider.currentCrashContext { [weak self] in
+            self?.inject(context: $0)
         }
 
         // Register for future `CrashContext` changes
         self.crashContextProvider.onCrashContextChange = { [weak self] in
-            self?.inject(currentCrashContext: $0)
+            self?.inject(context: $0)
         }
     }
 
     // MARK: - Interaction with `DatadogCrashReporting` plugin
 
     func sendCrashReportIfFound() {
-        queue.async {
-            self.plugin.readPendingCrashReport { [weak self] crashReport in
-                guard let self = self, let availableCrashReport = crashReport else {
-                    DD.logger.debug("No pending Crash found")
-                     self?.sender.send(launch: .init(didCrash: false))
-                    return false
-                }
-
-                DD.logger.debug("Loaded pending crash report")
-
-                guard let crashContext = availableCrashReport.context.flatMap({ self.decode(crashContextData: $0) }) else {
-                    // `CrashContext` is malformed and and cannot be read. Return `true` to let the crash reporter
-                    // purge this crash report as we are not able to process it respectively.
-                    self.sender.send(launch: .init(didCrash: true))
-                    return true
-                }
-
-                self.sender.send(report: availableCrashReport, with: crashContext)
+        // The plugin.readPendingCrashReport method is escaping, so
+        // every call in the closure must be thread-safe
+        plugin.readPendingCrashReport { [weak self] crashReport in
```

Comment:
blocking/ The removal of the `queue` causes the plugin work to be executed on the main thread ⚠️. For the default plugin, this will result in all I/O operations related to loading and purging crash data being performed on the main thread (see [here](https://github.com/DataDog/dd-sdk-ios/blob/ab8f2fbfb275e8ab7afc962c87794721854358fc/DatadogCrashReporting/Sources/PLCrashReporterIntegration/PLCrashReporterPlugin.swift#L39-L55)). That will likely cause perf hit and will impact app startup durations.

Additionally, by removing the queue we're breaking the contract expected by the plugin implementation. As specified [here](https://github.com/DataDog/dd-sdk-ios/blob/ab8f2fbfb275e8ab7afc962c87794721854358fc/DatadogCrashReporting/Sources/CrashReportingPlugin.swift#L19-L20), both plugin methods are expected to be called on a secondary thread:
```swift
    /// The SDK calls this method on a background thread. The implementation is free to choice any thread
    /// for executing the  `completion`.
    func readPendingCrashReport(completion: (DDCrashReport?) -> Bool)

    /// The SDK calls this method for each significant application state change.
    /// It is called on a background thread and succeeding calls are synchronized.
    func inject(context: Data)
```

Reply:
Ah yes, indeed! Good catch!